app/questionnaire/questionnaire_store_updater.py

```diff
@@ -82,5 +82,9 @@ def remove_answer_ids(self, answer_ids: List):
         self._questionnaire_store.add_or_update()
 
     def remove_completed_blocks(self, location):
-        self._questionnaire_store.remove_completed_blocks(location)
+        self._questionnaire_store.completed.blocks.remove(location)
+        self._questionnaire_store.add_or_update()
```

Comment:
Why do we need to call this now when we previously didn't?
We should check that this wont same the state multiple times per request as this has a substantial impact on performance.

Reply:
If we don't update at this point, a user that uses 'save and come back' on a block won't be directed to that block after signing in

Reply:
e.g. the test test_save_sign_out_complete_a_block_then_revisit_it fails